x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -383,18 +383,20 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                     .privileges("read", "write").build(),
                 // APM agent configuration
                 RoleDescriptor.IndicesPrivileges.builder()
-                    .indices(".apm-agent-configuration").privileges("all").build(),
+                    .indices(".apm-agent-configuration").privileges("all").allowRestrictedIndices(true).build(),
                 // APM custom link index creation
                 RoleDescriptor.IndicesPrivileges.builder()
-                    .indices(".apm-custom-link").privileges("all").build(),
+                    .indices(".apm-custom-link").privileges("all").allowRestrictedIndices(true).build(),
                 // APM telemetry queries APM indices in kibana task runner
                 RoleDescriptor.IndicesPrivileges.builder()
                     .indices("apm-*")
-                    .privileges("read", "read_cross_cluster").build(),
+                    .privileges("read", "read_cross_cluster")
+                    .allowRestrictedIndices(true).build(),
                 // Data telemetry reads mappings, metadata and stats of indices
                 RoleDescriptor.IndicesPrivileges.builder()
                     .indices("*")
-                    .privileges("view_index_metadata", "monitor").build(),
+                    .privileges("view_index_metadata", "monitor")
+                    .allowRestrictedIndices(true).build(),
```

Comment:
This change allows the `kibana_system` user to have some access to the `.security` index. I don't think this is what we want.

Reply:
I went through the list of system indices and pared down where I was allowing restricted indices to just the places where we are dealing with system indices, so I hope I've addressed this concern.

Reply:
Yes the change addressed my previous comment, thanks. But we still have another semantic change: before the #74212, this index privilege allow kibana to `monitor` and `view_index_metadata` for indices like `.watches`, `.triggered_watches`. These indices are now no longer accessible for `kibana_system`. Maybe this is the right thing to do. But I'd like to double check whether this is intentional.